### ISSUE TITLE
SKPayment.quantity member not accessible

### ISSUE THREAD

**Issue by cleofasv** [bizz84/SwiftyStoreKit#200]
### Platform

- [X] iOS
- [ ] macOS
- [ ] tvOS

### In app purchase type

- [X] Consumable
- [ ] Non-consumable 
- [ ] Auto-Renewable Subscription
- [ ] Non-Renewing Subscription

### Environment

- [X] Sandbox
- [X] Production

### Version
0.4.0

### Related issues
None

### Issue summary
The framework does not provide a way to access the SKPayment.quantity member during purchase. A workaround is to do multiple single purchase transactions to match the quantity, but this would generate multiple receipts.

### What did you expect to happen
Enable users to access SKPayment.quantity to support multiple purchases in one purchase transaction to receive just one transaction receipt.

### What happened instead
Could not find a way to access the SKPayment.quantity.

**Comment by bizz84** [comment]
I plan to implement this soon. The new API will likely look like this:

```swift
SwiftyStoreKit.purchaseProduct(productId, atomically: atomically, quantity: quantity)
```
I will also look into exposing the `quantity` parameter of `SKProduct` in the `Product` object returned by the purchase method.

**Comment by cleofasv** [comment]
Thanks so much @bizz84! I'll keep posted on future releases of SwiftyStoreKit. Also, if ever you need a hand on implementation, I'm very willing to contribute.

**Comment by bizz84** [comment]
@cleofasv I have added the `quantity` option when making purchases. This is also returned as part of `Payment` when the callback is called.

You can test this on the latest [develop](https://github.com/bizz84/SwiftyStoreKit/tree/develop) branch.

As for contributing, as soon as I can get some issues ready with some implementation specs it would be great to get you on board!

**Comment by cleofasv** [comment]
Thank you so much, @bizz84! I won't be able to test this just yet. Our code base is still Swift2.3, and migration to Swift3 is set to start on June/July. For now, I will be using the workaround.

**Comment by bizz84** [comment]
@cleofasv While I don't have bandwidth to support Swift 2.3, implementing this on your own fork is fairly straightforward:
* Locate the `startPayment()` method in `InAppProductPurchaseRequest`, and add the following line:
```swift
let payment = SKMutablePayment(product: product)
payment.applicationUsername = applicationUsername
payment.quantity = quantity // new code
```
* Add a `quantity` parameter to `InAppProductPurchaseRequest.startPayment()`
* Add a `quantity` parameter to `SwiftyStoreKit.purchase()`, and pass that to `InAppProductPurchaseRequest.startPayment()`

This should be all you need for the purchase path. Any compiler errors can be fixed by passing the`quantity` parameter to various places if missing.

If you also want to see the quantity in your callback, you could just modify `PurchaseResult` and pass along the quantity from your `SKProductTransaction` as needed:
```swift
public enum PurchaseResult {
        case Success(productId: String, quantity: Int)
        case Error(error: PurchaseError)
}
```

**Comment by bizz84** [comment]
Closing this issue as this is now available on pod version [0.9.0](https://github.com/bizz84/SwiftyStoreKit/releases/tag/0.9.0)

**Comment by cleofasv** [comment]
Oh alright @bizz84, we'll just for 2.3 and do our modifications there. Thank you so much for the support!
